### PR TITLE
Initial check-in of model

### PR DESCRIPTION
**Pull request by sbreingan**

### CODE REVIEW

File: src/main/java/gov/scot/payments/paymentinstruction/PaymentInstructionStatus.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    Accepted,
+    Rejected,
+    Failed_To_Clear,
+    Submitted,
+    Settled,
+    RecallRequested,
+    Recalled,
+    Invalid,
+    Routed,
+    Celared,
```
Comment (Steve-Conway) [inline src/main/java/gov/scot/payments/paymentinstruction/PaymentInstructionStatus.java]: Cleared?

Comment (sbreingan) [reply src/main/java/gov/scot/payments/paymentinstruction/PaymentInstructionStatus.java]: Fixed now

File: src/main/java/gov/scot/payments/customer/Service.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package gov.scot.payments.customer;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.Value;
+
+import java.util.Date;
+
+@Value
+@Builder(toBuilder = true)
```
Comment (Steve-Conway) [inline src/main/java/gov/scot/payments/customer/Service.java]: Do we not need @Getter also?

Comment (Steve-Conway) [reply src/main/java/gov/scot/payments/customer/Service.java]: OK ignore me.

File: build.gradle [hunk 3]
```diff
@@ -37,6 +37,10 @@ sourceCompatibility = 1.11
 targetCompatibility = 1.11
 
 dependencies {
+
+    implementation ('org.projectlombok:lombok:1.18.6')
```
Comment (jamesheward) [inline build.gradle]: im not sure implementation should be needed.

File: src/main/java/gov/scot/payments/customer/Service.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+public class Service {
+
+    private String bucket;
+    private Date createdAt;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/customer/Service.java]: LocalDateTime

File: src/main/java/gov/scot/payments/customer/Service.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    private String createdBy;
+    private String fileFormat;
+    private String uuid;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/customer/Service.java]: UUID

File: src/main/java/gov/scot/payments/paymentfile/PaymentFile.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Value;
+
+import java.util.Date;
+
+@Value
+@Builder(toBuilder = true)
+public class PaymentFile {
+
+    private PaymentFileStatus status;
+    private Service service;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/paymentfile/PaymentFile.java]: Just a string as ref to the id

File: src/main/java/gov/scot/payments/paymentfile/PaymentFile.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    private Date createdAt;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/paymentfile/PaymentFile.java]: LocalDateTime

File: src/main/java/gov/scot/payments/paymentfile/PaymentFile.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    private String createdBy;
+    private String hash;
+    private String uuid;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/paymentfile/PaymentFile.java]: UUID

File: src/main/java/gov/scot/payments/paymentfile/PaymentFile.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    private String name;
+    private String type;
+    private String url;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/paymentfile/PaymentFile.java]: URL type

File: src/main/java/gov/scot/payments/paymentfile/PaymentFile.java (hunk 8 above)
Comment (Steve-Conway) [inline src/main/java/gov/scot/payments/paymentfile/PaymentFile.java]: Is this not ````private  UUID id```` like PaymentInstruction?

File: src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
+package gov.scot.payments.paymentinstruction;
+
+import gov.scot.payments.paymentfile.PaymentFile;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java]: unused import

File: src/main/java/gov/scot/payments/paymentinstruction/Recipient.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Value;
+
+@Value
+@Builder(toBuilder = true)
+public class Recipient {
+
+    private int accountNumber;
+    private int sortCode;
+    private String address;
+    private String BIC;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/paymentinstruction/Recipient.java]: lower case

File: src/main/java/gov/scot/payments/paymentinstruction/Recipient.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    private String email;
+    private String IBAN;
```
Comment (jamesheward) [inline src/main/java/gov/scot/payments/paymentinstruction/Recipient.java]: lower case

File: src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+public class PaymentInstruction {
+
+    private Recipient recipient;
+    private String batchId;
+    private LocalDateTime createdAt;
+    private String createdBy;
+    private UUID id;
+    private Map<String,String> kvRef;
+    private String statusMessage;
+    private Money targetMoney;
```
Comment (Steve-Conway) [inline src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java]: ````targetAmount```` ?

File: src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+    private LocalDateTime latestTargetSettlementDate;
+    private LocalDateTime earliestTargetSettlementDate;
+
+    private String targetSettlementCurrency;
+    private String paymentFileId;
```
Comment (Steve-Conway) [inline src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java]: Should this be ````private PaymentFile fileReference````? In that case, does the translator need to be passed the fileReference by its caller?

File: src/main/java/gov/scot/payments/paymentinstruction/PaymentInstructionStatus.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
+package gov.scot.payments.paymentinstruction;
+
+public enum PaymentInstructionStatus {
+
+    New,
+    Vaidated,
```
Comment (Steve-Conway) [inline src/main/java/gov/scot/payments/paymentinstruction/PaymentInstructionStatus.java]: ````Validated````

File: src/main/java/gov/scot/payments/paymentinstruction/Recipient.java [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder(toBuilder = true)
+public class Recipient {
+
+    private int accountNumber;
+    private int sortCode;
```
Comment (Steve-Conway) [inline src/main/java/gov/scot/payments/paymentinstruction/Recipient.java]: Do we want a SortCode class here?

File: build.gradle (hunk 3 above)
Comment (sbreingan) [reply build.gradle]: I can't get it to work without implementation but changed to compileOnly. The latest version of Gradle (5.1.2) was causing IntelliJ to give exceptions so have left at version 4 for now.

File: src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java (hunk 14 above)
Comment (sbreingan) [reply src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java]: James suggested it should just be an id, but we can revisit when more of the structure is in place.

File: src/main/java/gov/scot/payments/paymentinstruction/Recipient.java (hunk 16 above)
Comment (sbreingan) [reply src/main/java/gov/scot/payments/paymentinstruction/Recipient.java]: As discussed we can add it if we need it.

File: src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java (hunk 13 above)
Comment (sbreingan) [reply src/main/java/gov/scot/payments/paymentinstruction/PaymentInstruction.java]: Changed.

**Pull request merged**